### PR TITLE
Add Bitcoin Core config options.

### PR DESCRIPTION
**Pull request by jlopp**
Add all the data mappings for config options and expose them in the UI via the Editor.

### CODE REVIEW

**Comment by Melvillian** [comment]
bitcoind does not accept boolean config values of "true" or "false" but rather 1 and 0. Replace line 367 of components/Editor.js

`onChange={(ev) => this.change(settings[section], prop)(ev.target.checked ? 1 : 0)}`

so that 0 and 1 will be written to the conf file.

File: src/data.json [hunk 1]
```diff
@@ -11,662 +11,800 @@
...
+    },
+    "usehd": {
+      "name": "Use HD Wallet",
+      "description": "Use hierarchical deterministic (BIP32) key generation. Only has effect during wallet creation/first start.",
+      "default": true
+    },
+    "walletrbf": {
+      "name": "Enable Replace By Fee Transactions",
+      "description": "Send transactions with full-RBF opt-in enabled.",
+      "default": true
```
Comment (Melvillian) [inline src/data.json]: default should be `false` since that's the default in 0.14 Core config

File: src/components/Editor.js [hunk 2]
```diff
@@ -24,124 +24,159 @@ class Editor extends Component {
...
+          { this.flag('debug', 'logtimestamps') }
+          { this.flag('debug', 'logtimemicros') }
+          { this.number('debug', 'mocktime') }
+          { this.number('debug', 'maxsigcachesize') }
+          { this.number('debug', 'maxtipage') }
+          { this.decimal('debug', 'maxtxfee') }
+          { this.flag('debug', 'printtoconsole') }
+          { this.text('debug', 'promiscuousmempoolflags') }
+          { this.flag('debug', 'shrinkdebugfile') }
+          { this.flag('debug', 'testnet') }
```
Comment (Melvillian) [inline src/components/Editor.js]: `testnet` and `regtest` should not be individual boolean options in the `debug` section, but should be a `.select` style option in the `core` section where they user can select [mainnet, testnet3, regtest, simnet] with mainnet (no option) as the default.

**Comment by jlopp** [comment]
Good catch; I updated that code to output the flag values and 0 and 1 and I updated all the json files to store default values as 0 and 1 rather than booleans.

File: src/components/Presets/Presets.js [hunk 3]
```diff
@@ -25,17 +32,16 @@ function mix (a, b) {
...
-  // Regtest node
-  // Non-syncing node
+  'Non-standard Ports': ports,
+  'Low Bandwidth': bandwidth,
+  'Low Disk Usage': pruned,
+  'Raspberry Pi': pi,
+  'Privacy': tor,
+  'Testnet': testnet,
+  'Regtest': regtest,
+  'Non-syncing': nosync
```
Comment (Melvillian) [inline src/components/Presets/Presets.js]: we need to add preset configs for nosync, right now it's empty.

I think all it needs is `connect = 0`

File: src/data.json [hunk 4]
```diff
@@ -11,662 +11,800 @@
...
+      "default": 0.00001,
+      "min": 0
+    },
+    "bytespersigop": {
+      "name": "Bytes Per Signature Operation",
+      "description": "Equivalent bytes per sigop in transactions for relay and mining.",
+      "default": 20,
+      "min": 1
+    },
+    "datacarrier": {
```
Comment (jlopp) [inline src/data.json]: Yeah, they are the same. Though no one ever calls them "data carrier" transactions during normal usage.

File: src/components/Editor.js (hunk 2 above)
Comment (jlopp) [inline src/components/Editor.js]: That would be preferable, though I don't think that the current logic makes it easy to support different variable names for the same selection list.

File: src/components/Presets/Presets.js (hunk 3 above)
Comment (jlopp) [inline src/components/Presets/Presets.js]: Maybe you're looking at an older diff; I added the nosync configs

**Comment by jlopp** [comment]
I'll change the section description lines to start with a '#' so that they are ignored.

You can find most of the config options in 2 places in Core:
https://github.com/bitcoin/bitcoin/blob/master/src/init.cpp#L333
and https://github.com/bitcoin/bitcoin/blob/f34cdcbd806d3e7c9d5dbac2a201755a4cc9828b/src/wallet/wallet.cpp#L3588

**Pull request merged**